grass7/imagery/i.gabor/i.gabor.html

```diff
@@ -0,0 +1,166 @@
+<h2>DESCRIPTION</h2>
+
+<p>
+<em>i.gabor</em> computes texture raster maps from a user-specified Gabor
+filter bank and raster map.
+The 2 dimensional Gabor filter is an orientation sensitive filter which mimics
+the cells within the human visual cortex by creating a filter bank of multiple
+orientations and scales to pick up different frequency responses.
+This mimicing or simulation of the visual cortex, and in particular its
+sensitivity to various orientations, sets the filter aside from other standard
+texture and edge detectors.
+Additionally, the Gabor filter has been shown to have high localization
+optimization/performance in both the spatial and frequency domains.
+The localized nature of the filter then allows for the parameters to be tweaked
+and changed to retrieve different features which suit the visual needs of the
+analysis.
+Subsequently, the Gabor filter has been used in hyperspectral image
+classification, water body and river extraction, land use classification and
+change detection, and is especially powerful when used with Object Based Image
+Analysis (OBIA) and edge detection algorithms.
+
+
+<h2>NOTES</h2>
+
+<em>i.gabor</em> uses <a href="https://numpy.org/">NumPy</a> to handle the
+entire raster map in memory.
+This can result in a bottleneck when processing large raster maps.
```

Comment:
I would be really great if there was an option to not treat everything in memory. Even though RAM is increases quickly, so does image size and resolution and we've always tried to make GRASS GIS robust to this (although some of the other filter addons do not). Would some sort of streaming approach be feasible ?

Reply:
I believe that would be very important to add, especially as an intention of  mine for this module was to use it with high resolution imagery. I believe it will be feasible with the GRASS RasterRow/Segement classes.